earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -74,7 +108,7 @@ DOMDocument *
 CreateEmptyDocument(const std::string &rootTagname) throw()
 {
   InitializeXMLLibrary();
-
+  if (doTerminate) pthread_mutex_lock(&xmlDocLock);
```

Comment:
For locking an unlocking, I think declaring xmlDocLock as a khMutex: `khMutex xmlDocLoc` and then calling `lock` or `unlock` as needed will work: `xmlDocLoc.loc(); /* other stuff */ xmlDocLoc.unlock();`